src/cargo/core/profiles.rs

```diff
@@ -470,6 +470,18 @@ impl ProfileMaker {
         unit_for: UnitFor,
     ) -> Profile {
         let mut profile = self.default;
+        if unit_for.is_for_host() {
+            // For-host units are things like procedural macros, build scripts, and
+            // their dependencies. For these units most projects simply want them
+            // to compile quickly and the runtime doesn't matter too much since
+            // they tend to process very little data. For this reason we default
+            // them to a "compile as quickly as possible" mode which for now means
+            // basically turning down the optimization level and avoid limiting
+            // codegen units. This ensures that we spend little time optimizing as
+            // well as enabling parallelism by not constraining codegen units.
+            profile.opt_level = InternedString::new("0");
+            profile.codegen_units = None;
```

Comment:
I think I would expect this to be in-between `merge_profile` and `merge_toml_overrides`.  Otherwise, if someone specifies:
```toml
[profile.release]
opt-level = 2
```
That would disable this change.  I would maybe expect something more like:

```rust
if let Some(toml) = &self.toml {
    merge_profile(&mut profile, toml);
}
if unit_for.is_for_host() {
    profile.opt_level = InternedString::new("0");
    profile.codegen_units = None;
}
if let Some(toml) = &self.toml {
    merge_toml_overrides(pkg_id, is_member, unit_for, &mut profile, toml);
}
```

does that make sense?

Reply:
Heh I'm glad at least one of us is intimately familiar with how features work.